src/renderer/dx/DxRenderer.cpp

```diff
@@ -1587,6 +1616,16 @@ CATCH_RETURN()
 
     RETURN_IF_FAILED(InvalidateAll());
 
+    if (_retroTerminalEffects && _d3dDeviceContext && _pixelShaderSettingsBuffer)
+    {
+        _ComputePixelShaderSettings();
+        try
+        {
+            _d3dDeviceContext->UpdateSubresource(_pixelShaderSettingsBuffer.Get(), 0, NULL, &_pixelShaderSettings, 0, 0);
+        }
+        CATCH_RETURN();

```

Comment:
Sorry, I don't know if my previous comment actually got saved on the backend b/c of the github 500 errors.

I think this should be `RETURN_IF_FAILED(_d3dContext->UpdateSubresource(x, y, z))`. UpdateSubresource won't throw an exception b/c it's on the other side of a COM barrier, it'll return an HRESULT (which right now we're discarding)

Reply:
`UpdateSubresource` returns a void. Should I do something else instead?

I added the `try` cuz the static analysis build was giving this error: `Error C26447: The function is declared 'noexcept' but calls function 'UpdateSubresource()' which may throw exceptions (f.6).`

Reply:
oh, that's interesting. meh. stupid static analyzer can't be taught that COM things don't throw exceptions. Fine, this is fine.